.github/workflows/src/arm-auto-signoff-preview.js

```diff
@@ -0,0 +1,173 @@
+// @ts-check
+
+import path from "path";
+import { extractInputs } from "../../src/context.js";
+import { LabelAction } from "../../src/label.js";
+import { getChangedSwaggerFiles } from "./changed-files.js";
+import { execRoot } from "./exec.js";
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @returns {Promise<LabelAction>}
+ */
+export default async function getLabelAction({ github, context, core }) {
+  let owner = process.env.OWNER || "";
+  let repo = process.env.REPO || "";
+  let issue_number = parseInt(process.env.ISSUE_NUMBER || "");
+  let head_sha = process.env.HEAD_SHA || "";
+
+  if (!owner || !repo || !issue_number || !head_sha) {
+    let inputs = await extractInputs(github, context, core);
+    owner = owner || inputs.owner;
+    repo = repo || inputs.repo;
+    issue_number = issue_number || inputs.issue_number;
+    head_sha = head_sha || inputs.head_sha;
+  }
+
+  return await getLabelActionImpl({
+    owner,
+    repo,
+    issue_number,
+    head_sha,
+    github,
+    core,
+  });
+}
+
+/**
+ * @param {Object} params
+ * @param {string} params.owner
+ * @param {string} params.repo
+ * @param {number} params.issue_number
+ * @param {string} params.head_sha
+ * @param {(import("@octokit/core").Octokit & import("@octokit/plugin-rest-endpoint-methods/dist-types/types.js").Api & { paginate: import("@octokit/plugin-paginate-rest").PaginateInterface; })} params.github
+ * @param {typeof import("@actions/core")} params.core
+ * @returns {Promise<LabelAction>}
+ */
+export async function getLabelActionImpl({
+  owner,
+  repo,
+  issue_number,
+  head_sha,
+  github,
+  core,
+}) {
+  const incrementalChangesToExistingRP =
+    await incrementalChangesToExistingResourceProvider(core);
+
+  if (!incrementalChangesToExistingRP) {
+    return LabelAction.Remove;
+  }
+
+  // TODO: Try to extract labels from context (when available) to avoid unnecessary API call
+  const labels = (
+    await github.rest.issues.listLabelsOnIssue({
+      owner: owner,
+      repo: repo,
+      issue_number: issue_number,
+    })
+  ).data.map((label) => label.name);
+
+  const allLabelsMatch =
+    labels.includes("ARMReview") &&
+    !labels.includes("NotReadyForARMReview") &&
+    labels.includes("ARMBestPractices") &&
```

Comment:
>labels.includes("ARMBestPractices") && [](http://example.com/codeflow?start=4&length=38)

Since this is a label that needs to be added manually by the author and we dont intend to ask authors to do this in preview mode, lets remove this for now.. Since the idea for preview is for us to check if all the main logic of this check is doing what its supposed to do, having this label requirement will not give us the coverage we need in preview. Lets add this back when we are ready to make auto signoffs generally available..